File: backend/app/models/conversation.py
from pydantic import BaseModel, ConfigDict


class Message(BaseModel):
    """A message in the conversation."""

    model_config = ConfigDict(extra="forbid")

    character: str  # message sender (characterId)
    content: str | None = None
    recipient: str  # sender's free-form label for the recipient
    thoughts: str
    mood: str  # Free-form mood description
    mood_emoji: str  # Matching emoji for the mood
    reaction_on_previous_message: str | None = None
    timestamp: str  # ISO format datetime
    unix_timestamp: float  # Unix timestamp (Epoch time)
    calculated_speaking_time: float  # seconds
    conversation_rating: int | None = None
    end_conversation: bool


class Conversation(BaseModel):
    """A conversation."""

    messages: list[Message]

File: turborepo/packages/database/src/schemas/conversation.ts
import z from 'zod';

export const MessageSchema = z.object({
  timestamp: z.number(),
  character: z.string(),
  recipient: z.string().describe('The recipient of the message - from the eyes of the character'),
  content: z.string().optional().nullable(),
  thoughts: z.string(),
  mood: z.string(),
  mood_emoji: z.string(),
  reaction_on_previous_message: z.string().optional().nullable(),
  unix_timestamp: z.number(),
  calculated_speaking_time: z.number().nullable(),
  conversation_rating: z.number().optional().nullable(),
  end_conversation: z.boolean().nullable(),
});
export type Message = z.infer<typeof MessageSchema>;

export const ConversationSchema = z.object({
  messages: z.array(MessageSchema),
});
export type Conversation = z.infer<typeof ConversationSchema>;
